ironic/tests/unit/drivers/modules/drac/test_raid.py

```diff
@@ -1380,48 +1406,47 @@ def test_create_configuration_fails_if_disk_already_reserved(
 
     @mock.patch.object(drac_common, 'get_drac_client', spec_set=True,
                        autospec=True)
-    @mock.patch.object(drac_raid, 'list_virtual_disks', autospec=True)
+    @mock.patch.object(drac_common, 'get_reboot_options', spec_set=True,
+                       autospec=True)
+    @mock.patch.object(drac_raid, 'list_raid_controllers', autospec=True)
     @mock.patch.object(drac_job, 'validate_job_queue', spec_set=True,
                        autospec=True)
     @mock.patch.object(drac_raid, 'commit_config', spec_set=True,
                        autospec=True)
-    @mock.patch.object(drac_raid, 'delete_virtual_disk', spec_set=True,
+    @mock.patch.object(drac_raid, 'reset_raid_config', spec_set=True,
                        autospec=True)
-    def test_delete_configuration(self, mock_delete_virtual_disk,
+    def test_delete_configuration(self, mock_reset_raid_config,
                                   mock_commit_config,
                                   mock_validate_job_queue,
-                                  mock_list_virtual_disks,
+                                  mock_list_raid_controllers,
+                                  mock_get_reboot_options,
                                   mock_get_drac_client):
         mock_client = mock.Mock()
         mock_get_drac_client.return_value = mock_client
```

Comment:
I believe `mock_get_reboot_options`'s return value needs to be set to an object that looks like `dracclient.constants.RebootRequired`, analogous to how `mock_get_drac_client`'s is here. The following worked for me.

    mock_get_reboot_options.return_value = DictToObj(
        {'true': 'true', 'optional': 'optional', 'false': 'false'})

Reply:
Thanks richard for resolving the issue. it passed the testcases.
Updated the change with latest commit.